### PR TITLE
RIFEX-65 Add nifty component for warning and errors

### PR DESCRIPTION
**Pull request by JONAF2103**
This PR add the nifty warning and error components to our pages:
The warning/error component:
imagen
The toast or message component:
imagen

In this PR we are:

* Fixing a bug on the nifty warning component
* Fixing a bug on the nifty error component
* Adding both to our page reducer
* Adding the navbar for our pages
* Adding support for navigation on the navbar
* Updating our screens to support navigation bar

This affects the ui for nifty and rif components

### CODE REVIEW

File: ui/app/rif/pages/domainsDetailPage/domainsDetailPage.js [hunk 1]
```diff
@@ -40,9 +32,7 @@ function mapStateToProps (state) {
 }
 
 const mapDispatchToProps = dispatch => {
-	return {
-		goBack: () => dispatch(rifActions.navigateTo(pageNames.rns.domains)),
-	}
+	return {}
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domainsDetailPage/domainsDetailPage.js:35]: If we are not using any action binded with redux, we can just omit this. `connect ` accepts passing only `mapStateToProps`

File: ui/app/rif/pages/index.js [hunk 2]
```diff
@@ -27,34 +31,87 @@ const pageNames = {
   },
 }
 
-function buildScreen (screenComponent, currentView) {
-  return h(screenComponent, {key: currentView.screenName})
+function buildScreen (screenComponent, context, dispatch) {
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/index.js:34]: This is great. 

We should think about generalize this more. 

For example, the following code: 

```
if (context.params.showDomainsSearch) {
    searchDomains = (<SearchDomains/>);
  }
```

relates to RNS stuff, but we can make this component more abstract, treating that as a header or generic searchbar. Imagine that in a future we have other screens that have the same structure but are not RNS related.

**Review by marcosmartinez7 — APPROVED** [review]

File: ui/app/rif/pages/domainsDetailPage/domainsDetailPage.js (hunk 1 above)
Comment (JONAF2103) [reply ui/app/rif/pages/domainsDetailPage/domainsDetailPage.js:35]: I think that it's going to be changed by @rorolopetegui so i guess it's not going to be empty

File: ui/app/rif/pages/index.js (hunk 2 above)
Comment (JONAF2103) [reply ui/app/rif/pages/index.js:34]: Yeah maybe we should create a new component called header or something like that and then inside put the search bar probably, we should make that more generic in the future, i agree, we can refactor that later if we need to.

File: ui/app/rif/actions.js [hunk 3]
```diff
@@ -193,14 +185,44 @@ function showMenu (data) {
   }
 }
 
+function navigateBack () {
```
Comment (rorolopetegui) [inline ui/app/rif/actions.js:196]: This was so necessary! Nice implementation

File: ui/app/rif/pages/domainsDetailPage/domainsDetailPage.js (hunk 1 above)
Comment (rorolopetegui) [reply ui/app/rif/pages/domainsDetailPage/domainsDetailPage.js:35]: Yep, in next PRs we're going to use the Connect with botch parameters

**Review by rorolopetegui — APPROVED** [review]
lgtm

File: ui/app/rif/actions.js (hunk 3 above)
Comment (JONAF2103) [reply ui/app/rif/actions.js:196]: Thanks man!

**Pull request merged**